encoders/protoc-gen-firebase-encoders/src/main/kotlin/com/google/firebase/encoders/proto/codegen/Parsing.kt

```diff
@@ -27,9 +33,128 @@ interface DescriptorParser {
     fun parse(protoFiles: List<FileDescriptorProto>): Collection<UserDefined>
 }
 
+/**
+ * Default [DescriptorParser] implementation.
+ *
+ * The parser repackages all input messages into the `vendorPackage` of the provided [config].
+ * Additionally it only returns messages that are marked for inclusion in [config].
+ *
+ * Any extensions that are encountered are added directly to extended messages as fields, so
+ * extensions are indistinguishable from message's own fields.
+ */
 class DefaultParser @Inject constructor(private val config: CodeGenConfig) : DescriptorParser {
-    override fun parse(protoFiles: List<FileDescriptorProto>): Collection<UserDefined> {
-        return listOf()
+    override fun parse(files: List<FileDescriptorProto>): List<UserDefined> {
+        val parsedMessages = files.asSequence().flatMap { file ->
+            val javaPackage = "${config.vendorPackage}.${file.`package`}"
+
+            val parent = Owner.Package(file.`package`, javaPackage, file.name)
+            parseEnums(parent, file.enumTypeList).plus(parseMessages(parent, file.messageTypeList))
+        }.toList()
+
+        val extensions = discoverExtensions(files)
+
+        val messagesWithExtensions = parsedMessages.map { msg ->
+            val msgExtensions = extensions[msg.protobufFullName]
+            if (msg !is Message || msgExtensions == null) {
```

Comment:
if msg isn't Message, wouldn't the protobufFullName above fail before reaching this point? Also

Reply:
It wouldn't fail, because if it's not a `Message` then it's a `ProtoEnum` which also has the `protobufFullName` property.

I renamed the variables to make it more clear